tests/test_update_file_manager.py

```diff
@@ -71,3 +77,792 @@ def test_check_if_update_based_on_last_modified(database, mocker, mock_db):
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    ip = "1.2.3.4"
+    blacklist = "test_blacklist.txt"
+    update_manager.add_to_ip_ctr(ip, blacklist)
+    assert update_manager.ips_ctr[ip] == {
+        "times_found": 1,
+        "blacklists": ["test_blacklist.txt"],
+    }
+
+
+def test_parse_ti_feed_valid_data(mocker, mock_db, tmp_path):
+    """
+    Test parse_ti_feed with valid data containing both IP and domain.
+    """
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    update_manager.url_feeds = {
+        "https://example.com/test.txt": {
+            "threat_level": "low",
+            "tags": ["tag3"],
+        }
+    }
+    test_data = """# Comment
+    1.2.3.4,Test description
+    example.com,Another description
+    """
+    test_file = tmp_path / "test.txt"
+    test_file.write_text(test_data, encoding="utf-8")
+    mocker.patch("builtins.open", mock_open(read_data=test_data))
+    result = update_manager.parse_ti_feed(
+        "https://example.com/test.txt", str(test_file)
+    )
+    mock_db.add_ips_to_IoC.assert_any_call(mocker.ANY)
+    mock_db.add_domains_to_IoC.assert_any_call({"example.com": mocker.ANY})
+    assert result is True
+
+
+def test_parse_ti_feed_invalid_data(mocker, mock_db, tmp_path):
+    """Test parse_ti_feed with invalid data."""
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    update_manager.url_feeds = {
+        "https://example.com/invalid.txt": {
+            "threat_level": "low",
+            "tags": ["tag3"],
+        }
+    }
+    test_data = """# Comment
+    invalid_data,Description
+    """
+    test_file = tmp_path / "invalid.txt"
+    test_file.write_text(test_data, encoding="utf-8")
+    mocker.patch("builtins.open", mock_open(read_data=test_data))
+    result = update_manager.parse_ti_feed(
+        "https://example.com/invalid.txt", str(test_file)
+    )
+    mock_db.add_ips_to_IoC.assert_not_called()
+    mock_db.add_domains_to_IoC.assert_not_called()
+    assert result is False
+
+
+@pytest.mark.parametrize(
+    "file_content, cached_hash, expected_result",
+    [
+        ("test content", {}, True),  # Testcase1 New file
+        ("new content", "old_hash", True),  # Testcase2 Updated file
+        # Testcase3 Unchanged file
+        (
+            "test content",
+            "e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+            True,
+        ),
+    ],
+)
+def test_check_if_update_org(
+    mock_db, tmp_path, file_content, cached_hash, expected_result
+):
+    """Test check_if_update_org with different file and cache scenarios."""
+    update_manager = ModuleFactory().create_update_manager_obj(mock_db)
+    test_file = tmp_path / "test_org"
+    test_file.write_text(file_content, encoding="utf-8")
+    mock_db.get_TI_file_info.return_value = {"hash": cached_hash}
+    result = update_manager.check_if_update_org(str(test_file))
+    assert result is expected_result
+
```

Comment:
here i would mock get_TI_file_info() and utils.get_hash_from_file() instead of creating a file and writing to it